CMakeLists.txt

```diff
@@ -33,9 +56,43 @@ endif()
 add_subdirectory(extlibs/minizip)
 add_subdirectory(extlibs/elzip)
 
-# SFML 2.5 is minimum version
-find_package(SFML 2.5 COMPONENTS system window graphics network audio REQUIRED)
-# SFML is required by sfe (SFml Extensions)
+# snippet taken from
+# https://github.com/microsoft/vcpkg/blob/master/ports/sfml/usage
+if(OBE_USE_VCPKG)
+    include(FindLua)
+
+    find_package(lua REQUIRED)
+    find_package(sol2 CONFIG REQUIRED)
+    find_package(fmt CONFIG REQUIRED)
+    find_package(spdlog CONFIG REQUIRED)
+    find_package(Catch2 CONFIG REQUIRED)
+    find_path(TINYDIR_INCLUDE_DIRS "tinydir.h")
+
+    find_package(SFML COMPONENTS system window graphics CONFIG REQUIRED)
+
+    include_directories(${TINYDIR_INCLUDE_DIRS})
+
+    if(WIN32)
+        link_libraries(FLAC OpenAL OpenGL Vorbis)
+    else()
+        link_libraries(X11 FLAC UDev OpenAL)
+    endif()
+
+
+    link_libraries(${LUA_LIBRARIES})
+    include_directories(${LUA_INCLUDE_DIR})
+
+    link_libraries(sol2::sol2)
```

Comment:
Can we use `target_link_libraries` instead ?

Reply:
That is what I initially used, but I got an error saying that the target wasn't built by this CMakeLists.txt

Reply:
Yup, the `target_link_libraries` are done in a separate [`CMakeLists.txt`](https://github.com/Sygmei/ObEngine/blob/master/src/Core/CMakeLists.txt#L26) (which is the CMakeLists.txt for ÖbEngineCore).

Reply:
Hmm.. How can we make the other `CMakeLists.txt` know that we are using vcpkg?

Reply:
Variables are shared between the `CMakeLists.txt` so `OBE_USE_VCPKG` should be accessible from there